src/directives/for-of/for_of.directive.ts

```diff
@@ -358,14 +378,25 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
         this.applyPageSizeChange();
         if (this.igxForOf && this.igxForOf.length && this.dc) {
             const embeddedViewCopy = Object.assign([], this._embeddedViews);
-            const endingIndex = this._pageSize + this._currIndex;
-            for (let i = this._currIndex; i < endingIndex && this.igxForOf[i] !== undefined; i++) {
-                const input = this.igxForOf[i];
-                const embView = embeddedViewCopy.shift();
-                const cntx = (embView as EmbeddedViewRef<any>).context;
-                cntx.$implicit = input;
-                cntx.index = this.igxForOf.indexOf(input);
+            if (!this.igxForRemote) {
+                const endingIndex = this._pageSize + this._currIndex;
+                for (let i = this._currIndex; i < endingIndex && this.igxForOf[i] !== undefined; i++) {
+                    const input = this.igxForOf[i];
```

Comment:
the two for loops seem to differ very slightly and their bodies are identical, therefore this can be simplified by applying borders of the for loop by the value of igxForRemote and leave only one for loop